### PR TITLE
Workaround for incompatible serialization for `Inf` floats in Hive text write

### LINKED ISSUE #7686: [BUG] Large floating point values not written as "Inf" in Hive text writer
The following floating point values are not written as `Inf` by the Hive delimited text writer in `spark-rapids`:
```
3.4028236E38
3.4028236e+38
1.7976931348623157E99
1.7976931348623157e+99
1.7976931348623157E308
1.7976931348623157e+308
```
These values are reported as `Inf` on Apache Spark 3.2.x, on CPU.

The values are captured in [`integration_tests/src/test/resources/hive-delim-text/simple-float-values/simple_float_values.hive.txt`](https://github.com/NVIDIA/spark-rapids/blob/branch-23.02/integration_tests/src/test/resources/hive-delim-text/simple-float-values/simple_float_values.hive.txt). The test that reads from this file is left as an `xfail` test. It would be good to fix the parser for `Inf` values.

### PR DESCRIPTION
**Pull request by mythrocks**
Fixes #7686.
Depends on #7708.

The `GpuHiveTextFileFormat` is currently serializing Float(Infinity) as 'Inf' instead of
'Infinity'. (It boils down to the CUDF implementation using 'Inf'.)
    
Apache Spark (and Apache Hive) expect Infinity strings to be represented as 'Infinity'. 'Inf' is read as `NULL`.
    
This commit adds special handling for floating point, such that 'Inf' is replaced with
'Infinity' before serializing to the table.
    
Signed-off-by: MithunR <mythrocks@gmail.com>

### CODE REVIEW

**Comment by mythrocks** [comment]
Note: This PR includes the changes introduced in #7708. It should only be merged after #7708 is merged.

**Review by revans2 — APPROVED** [review]
I am not thrilled with stringReplace, but it is fast and does what we want.

**Comment by mythrocks** [comment]
> I am not thrilled with stringReplace, ...

I agree. This and the timestamp reformatting, both feel as though they should be configurable in `libcudf`. I'll see about this in `libcudf`, in the next sprint, if I'm allowed.

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
I ran some tests to check the impact of the additional `Inf` checks, and string manipulations.

Transcribing 1 Billion floats with 0.1% `Inf` values does not seem to take any longer with the fix than without. (GPU: RTX 6000). About 8 seconds.

(For reference, the same, when run on CPU (Intel 10980XE: 36 threads) takes 28 seconds.)

**Comment by mythrocks** [comment]
Hmm. The logs indicate that all the `hive_delimited_text_test.py` tests are passing (apart from the xfails). I don't see failures at all. Here's the error at the bottom:
```
2023-02-09T01:40:01.9163059Z [2023-02-09T01:38:15.678Z] INTERNALERROR> >           assert False, formatted_error
2023-02-09T01:40:01.9163478Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E           AssertionError: Traceback (most recent call last): 
2023-02-09T01:40:01.9164003Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/_pytest/main.py", line 268, in wrap_session
2023-02-09T01:40:01.9164536Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 config.hook.pytest_sessionstart(session=session)
2023-02-09T01:40:01.9165049Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/pluggy/_hooks.py", line 265, in __call__
2023-02-09T01:40:01.9165545Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
2023-02-09T01:40:01.9166046Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/pluggy/_manager.py", line 80, in _hookexec
2023-02-09T01:40:01.9166516Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
2023-02-09T01:40:01.9167101Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 60, in _multicall
2023-02-09T01:40:01.9167468Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 return outcome.get_result()
2023-02-09T01:40:01.9167986Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/pluggy/_result.py", line 60, in get_result
2023-02-09T01:40:01.9168355Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 raise ex[1].with_traceback(ex[2])
2023-02-09T01:40:01.9168869Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 39, in _multicall
2023-02-09T01:40:01.9169231Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 res = hook_impl.function(*args)
2023-02-09T01:40:01.9169997Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/home/jenkins/agent/workspace/jenkins-rapids_premerge-github-6638-ci-2/integration_tests/src/main/python/spark_init_internal.py", line 115, in pytest_sessionstart      
2023-02-09T01:40:01.9170324Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 import pyspark
2023-02-09T01:40:01.9171073Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/home/jenkins/agent/workspace/jenkins-rapids_premerge-github-6638-ci-2/.download/spark-3.1.1-bin-hadoop3.2/python/pyspark/__init__.py", line 53, in <module>
2023-02-09T01:40:01.9171466Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 from pyspark.rdd import RDD, RDDBarrier
2023-02-09T01:40:01.9172203Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/home/jenkins/agent/workspace/jenkins-rapids_premerge-github-6638-ci-2/.download/spark-3.1.1-bin-hadoop3.2/python/pyspark/rdd.py", line 34, in <module> 
2023-02-09T01:40:01.9172646Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 from pyspark.java_gateway import local_connect_and_auth
2023-02-09T01:40:01.9173421Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E               File "/home/jenkins/agent/workspace/jenkins-rapids_premerge-github-6638-ci-2/.download/spark-3.1.1-bin-hadoop3.2/python/pyspark/java_gateway.py", line 29, in <module>
2023-02-09T01:40:01.9173954Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E                 from py4j.java_gateway import java_import, JavaGateway, JavaObject, GatewayParameters
2023-02-09T01:40:01.9174362Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E             ModuleNotFoundError: No module named 'py4j'
2023-02-09T01:40:01.9174661Z [2023-02-09T01:38:15.678Z] INTERNALERROR> E           assert False
2023-02-09T01:40:01.9174885Z [2023-02-09T01:38:15.678Z] INTERNALERROR> 
```

**Comment by mythrocks** [comment]
Build

**Comment by mythrocks** [comment]
Hi, @revans2. I seem to have dismissed your approval when I fixed the import order. Could I please bug you to approve again?

**Review by revans2 — APPROVED** [review]

**Pull request merged**
